### PR TITLE
Added Basic Authorization for UrlResource

### PR DESCRIPTION
**Pull request by x-x-z**
This pull request adds Basic Authorization for UrlResource.
This is fix for https://github.com/spring-projects/spring-boot/issues/6138

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@x-x-z Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=1822)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=1822) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@x-x-z Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=1822)!

File: spring-core/src/main/java/org/springframework/core/io/UrlResource.java [hunk 1]
```diff
@@ -167,6 +169,12 @@ private URL getCleanedUrl(URL originalUrl, String originalPath) {
 	public InputStream getInputStream() throws IOException {
 		URLConnection con = this.url.openConnection();
 		ResourceUtils.useCachesIfNecessary(con);
+
+		if (this.url.getUserInfo() != null) {
+			String basicAuth = "Basic " + new String(Base64Utils.encode(url.getUserInfo().getBytes()));
```
Comment (dreis2211) [inline spring-core/src/main/java/org/springframework/core/io/UrlResource.java]: What about using `Base64Utils.encodeToString()`?

Comment (x-x-z) [reply spring-core/src/main/java/org/springframework/core/io/UrlResource.java]: @dreis2211 
Thank you for your comment.
I fixed it in "005911e"

**Comment by dstepanov** [comment]
👍 
It would be nice if this change has been merged.

**Comment by snicoll** [comment]
@x-x-z can you share how this is a fix for the issue you've linked?

**Pull request closed without merging**